app/models/labware_creators/multi_stamp_tubes_using_tube_rack_scan.rb

```diff
@@ -148,6 +149,25 @@ def tubes_must_have_active_requests_of_expected_type
       end
     end
 
+    # Validates that the tubes in the parent_tubes hash contains at the source tube.
+    # If the provided file does not contain the source tube, an error message is added to the errors object.
+    def tubes_must_contain_source_tube
+      return unless file_valid?
+
+      # parent tube should be LRC Bank Seq or LRC Bank Spare
+      parent_tube_barcode = labware.barcode.prefix + labware.barcode.number.to_s
+      contains_source_tube =
+        parent_tubes.any? do |foreign_barcode, tube_in_db|
+          tube_in_db.present? && parent_tube_barcode == foreign_barcode
+        end
+
+      return if contains_source_tube
+      errors.add(
+        :base,
+        "Uploaded tube rack scan file does not contain the source tube for this tube rack (#{parent_tube_barcode})"
```

Comment:
Doesn't read quite right to me 🤔 what about:

```suggestion
        "Uploaded tube rack scan file does not contain the tube scanned on the previous page (#{parent_tube_barcode})"
```